### PR TITLE
Stop disabling known host files for handle_host_key, fix test_known_host_good

### PR DESCRIPTION
**Pull request by martinpitt**
See the commit descriptions for details.

### CODE REVIEW

**Comment by martinpitt** [comment]
@allisonkarlitskaya [That behaviour](https://github.com/allisonkarlitskaya/ferny/pull/15#pullrequestreview-1446554556) isn't very useful for our purposes, but I didn't find a better one in the SSH docs. Good that we have a plan how to get rid of it :sweat_smile:

File: test/test_basic.py [hunk 1]
```diff
@@ -213,9 +213,10 @@ async def test_correct_passphrase(self, key_dir: pathlib.Path, runtime_dir: path
         assert 'keys/id_rsa.enc' in prompt
 
     @pytest.mark.asyncio
-    @pytest.mark.xfail
     async def test_known_host_good(self, key_dir: pathlib.Path, runtime_dir: pathlib.Path) -> None:
-        await self.run_test(key_dir, runtime_dir, ZeroDivisionError(), 'passphrase',
+        # this calls do_hostkey() for the already known key, just in case it wants to supply additional keys
+        # don't do this and don't accept any, just rely on the existing one
+        await self.run_test(key_dir, runtime_dir, False, 'passphrase',
                             handle_host_key=True, known_host_key='scan')
```
Comment (martinpitt) [inline test/test_basic.py:220]: For actually fixing cockpit, I have to get rid of setting `{Global,User}KnownHostsFile=none` with handle_host_keys=True, as they ruin the show. Dropping them does not break any test.

However, *this* test here really ought to fail right now, as without any existing database there is no way that the SSH client can change an existing host key. I think this somehow interferes with setting another known_hosts file in the test, as *that's* the one that has the bad key. This does not model reality well, so let's fix this. Back to draft.

Comment (martinpitt) [reply test/test_basic.py:220]: Added a separate commit for that.

File: src/ferny/session.py [hunk 2]
```diff
@@ -139,8 +139,6 @@ async def connect(self,
         if handle_host_key and has_feature('KnownHostsCommand'):
             args.extend([
                 '-o', f'KnownHostsCommand={askpass_path} %I %H %t %K %f',
-                '-o', 'GlobalKnownHostsFile=none',
-                '-o', 'UserKnownHostsFile=none',
```
Comment (martinpitt) [inline src/ferny/session.py:143]: Turns out we don't strictly require this for cockpit with the current API, see https://github.com/cockpit-project/cockpit/pull/18856 . I still find it conceptually cleaner, though.

Comment (allisonkarlitskaya) [reply src/ferny/session.py:143]: I think the conceptual idea here was that if we have `handle_host_key` then we're taking over *all* control of handling the host key.

Part of my thinking here is that we're overriding the `KnownHostsCommand`.  At some point, that might have an impact on some system whereby we can no longer connect to a host because we don't gets its IPA key (or whatever).  Doing the same for the "files" approach was meant to make it clear that this behaviour is intended.

The other reason I did that was that, in cockpit, by the time we decide to run ferny in this mode, we know that we want the connection to fail with the hostkey in the `close` message.  This guarantees that it fails (since we return False).

All that being said, it counter-acts the documentation on the `do_host_key` callback, which states:

> Returning False means that ssh implements its default logic.

In short, I'm ambivalent about this change.  If you read the above and are still convinced that the change is correct, let's do it.  But in any case: one way or the other, this code is probably not going to survive the "big API change" we've been talking about.

Comment (martinpitt) [reply src/ferny/session.py:143]: > one way or the other, this code is probably not going to survive the "big API change" we've been talking about.

I'm counting on this! :grin: This is tricky to pin down in a consistent way, and I'm really looking forward to dropping both `KnownHostsCommand` and these options, and just let SSH do its thing. For that reason I also don't have a strong opinion about this commit. As you wrote (and my cockpit PR confirms), we don't need it for Cockpit, so it's not on the critical path. I just feel uneasy about it as the unit tests cannot actually cover the effect of these, as they set `UserKnownHostsFile` to a real file, and assert that `handle_known_hosts=True` will actually be able to tell the cases apart. So I'm still leaning towards landing it, as it will make the unit tests more "honest" and useful.

Comment (allisonkarlitskaya) [reply src/ferny/session.py:143]: Fine by me.

**Review by allisonkarlitskaya — APPROVED** [review]
